packages/core/src/virtual-module/virtual-routes.ts

```diff
@@ -80,17 +80,28 @@ export function createVirtualRouteComponent(route: VirtualRoute): string {
   return `
 ---
 import StoryPage from 'astrobook/pages/story.astro'
-import { isAstroStory } from 'astrobook/client'
+import { isAstroStory, isAstroComponentFactory } from 'astrobook/client'
 import * as m from '${route.storyModule.importPath}'
 
 const isAstro = isAstroStory(m)
 ---
 
 <StoryPage story={'${route.props.story}'} hasSidebar={${route.props.hasSidebar}}>
   {
-    isAstro
+    (({...m['${route.story.name}']}?.decorators || []).slice().reverse().reduce((carry, v) => {
+      const Dec = v()
+
+      return isAstroComponentFactory(Dec)
+        ? <Dec children={carry} />
+        : (
+          <Dec>
+            {carry}
+          </Dec>
+        )
```

Comment:
`<Dec>{carry}</Dec>` should just work for both Astro and other UI frameworks (React, Vue etc)